pkg/msr/msr_linux.go

```diff
@@ -206,13 +208,34 @@ func (m MSR) TestAndSet(c CPUs, clearMask uint64, setMask uint64) []error {
 			// We write only if there is a change. This is to avoid
 			// cases where we try to set a lock bit again, but the bit is
 			// already set
-			if n != v {
+			if n != v && set {
 				return binary.Write(port, binary.LittleEndian, n)
 			}
+			if n != v {
+				return fmt.Errorf("%#x", v)
```

Comment:
Include more information in the error, such as the CPU.

Reply:
That's going to be too verbose. The caller can do that. This is internal.